[Cocoa] Replace bool parameter to preparePlatformFont() with a 2-state enum

litherum opened the pull request:
#### 999e6580a40d042a8ee57e24284005f299448cdf
<pre>
[Cocoa] Replace bool parameter to preparePlatformFont() with a 2-state enum
<a href="https://bugs.webkit.org/show_bug.cgi?id=252232">https://bugs.webkit.org/show_bug.cgi?id=252232</a>
rdar://105439787

Reviewed by Tim Nguyen.

WebKit style prefers 2-state enums for parameters instead of bools.

* Source/WebCore/platform/graphics/cocoa/FontCacheCoreText.cpp:
(WebCore::preparePlatformFont):
(WebCore::fontWithFamily):
* Source/WebCore/platform/graphics/cocoa/FontCacheCoreText.h:

Canonical link: <a href="https://commits.webkit.org/260250@main">https://commits.webkit.org/260250@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/786d52ea0b09323f11dacadc497c61f368068792

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/107680 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/16732 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/40575 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/116827 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/116221 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/111570 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/18154 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/8050 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/99853 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/113433 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/13705 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/96900 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/41384 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/95621 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/28538 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/83219 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/9713 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/29887 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/10391 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/6773 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/15867 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/49472 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/7085 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/11951 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Source/WebCore/platform/graphics/cocoa/FontCacheCoreText.h:
```diff
@@ -63,7 +63,11 @@ struct VariationDefaults {
+enum class ApplyWeightWidthSlopeVariations : uint8_t {
+    Yes,
+    No,
+};
```
```suggestion
enum class ApplyWeightWidthSlopeVariations : bool {
    No,
    Yes,
};
```

Aside from this nit, the name `ApplyWeightWidthSlopeVariations` feels a bit clunky, I don't have any better suggestions, so I'm fine landing as is.

Maybe consider simplifying the name by transforming to something like `SlopeVariations::{ None, WeightWidth }` (not sure if it makes sense with the current code, but maybe you have better ideas).

litherum replied:
Yeah, this means "apply the weight variations, width variations, and the slope variations"

Probably "traits variations" is better - "traits" is a term-of-art

litherum committed:
[Cocoa] Replace bool parameter to preparePlatformFont() with a 2-state enum
https://bugs.webkit.org/show_bug.cgi?id=252232
rdar://105439787

Reviewed by Tim Nguyen.

WebKit style prefers 2-state enums for parameters instead of bools.

* Source/WebCore/platform/graphics/cocoa/FontCacheCoreText.cpp:
(WebCore::preparePlatformFont):
(WebCore::fontWithFamily):
* Source/WebCore/platform/graphics/cocoa/FontCacheCoreText.h:

Canonical link: https://commits.webkit.org/260250@main
